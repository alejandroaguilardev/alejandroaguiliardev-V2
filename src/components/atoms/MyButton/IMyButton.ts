export interface IMyButton {
	text: string;
}
