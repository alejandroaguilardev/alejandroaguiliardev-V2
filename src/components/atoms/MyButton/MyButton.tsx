import { IMyButton } from "./IMyButton"
import { Button, Typography } from "@mui/material"

export const MyButton = ({ text }: IMyButton) => {
    return (
        <Button
            sx={{
                backgroundColor: 'primary.main',
                color: '#fff',
                py: 1,
                px: 2,
                textTransform: 'capitalize',
                transition: ".5s backgroundColor",
                "&:hover": {
                    backgroundColor: 'secondary.main',
                }
            }}
        >
            <Typography sx={{ fontSize: ".7rem"}}>{text}</Typography>
        </Button>
    )
}
