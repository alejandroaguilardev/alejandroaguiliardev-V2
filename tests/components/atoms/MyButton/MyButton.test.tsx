import { render, screen } from "@testing-library/react";
import { MyButton } from "../../../../src/components/atoms/MyButton/MyButton";

describe("MyButton Test", () => {
  const text = "Button Test";
  test("toMacthSnapshot", () => {
    const { container } = render(<MyButton text={text} />);

    expect(container).toMatchSnapshot();
  });

  test("Props ", () => {
    render(<MyButton text={text} />);

    expect(screen.getByText(text)).toBeTruthy();
  });

});
