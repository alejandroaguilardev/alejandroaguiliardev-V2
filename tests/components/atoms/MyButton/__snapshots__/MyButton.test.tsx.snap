// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyButton Test toMacthSnapshot 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-xlnruk-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-15esv0h-MuiTypography-root"
    >
      Button Test
    </p>
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;
